controllers/azurejson_machine_controller.go

```diff
@@ -60,13 +62,35 @@ func (r *AzureJSONMachineReconciler) SetupWithManager(ctx context.Context, mgr c
 	)
 	defer done()
 
-	return ctrl.NewControllerManagedBy(mgr).
+	azureMachineMapper, err := util.ClusterToObjectsMapper(r.Client, &infrav1.AzureMachineList{}, mgr.GetScheme())
+	if err != nil {
+		return errors.Wrap(err, "failed to create mapper for Cluster to AzureMachines")
+	}
+
+	c, err := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
 		For(&infrav1.AzureMachine{}).
 		WithEventFilter(filterUnclonedMachinesPredicate{log: log}).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(log, r.WatchFilterValue)).
 		Owns(&corev1.Secret{}).
-		Complete(r)
+		Build(r)
+
+	if err != nil {
+		return errors.Wrap(err, "failed to create controller")
+	}
+
+	// Add a watch on Clusters to requeue when the infraRef is set. This is needed because the infraRef is not initially
+	// set in Clusters created from a ClusterClass.
+	if err := c.Watch(
```

Comment:
@CecileRobertMichon Do we want to add a watch to the AzureJSONMachineController as well? It also returns w/o an error if the infrastructureRef is nil.

Reply:
we need this on every azurejson controller (I believe you're missing machinepool)